torch/csrc/jit/runtime/jit_exception.h

```diff
@@ -11,5 +12,18 @@ struct TORCH_API JITException : public std::runtime_error {
   explicit JITException(const std::string& msg);
 };
 
+struct TORCH_API CustomJITException : public JITException {
```

Comment:
Can we enhance `JITException` to have an `optional<str>` field instead of subclassing? Edge use cases are sensitive to binary size increase and runtime overhead, and subclassing would add to both.